xbrowse/datastore/elasticsearch_datastore.py

```diff
@@ -596,18 +617,7 @@ def get_elasticsearch_variants(
                     'topmed_Hom': float(hit["topmed_Hom"] or 0) if "topmed_Hom" in hit else None,
                     'topmed_AN': float(hit["topmed_AN"] or 0) if "topmed_AN" in hit else None,
                 },
-                'db_freqs': {
-                    'AF': float(hit["AF"] or 0.0) if "AF" in hit else None,
-                    '1kg_wgs_AF': float(hit["g1k_AF"] or 0.0) if "g1k_AF" in hit else None,
-                    '1kg_wgs_popmax_AF': float(hit["g1k_POPMAX_AF"] or 0.0) if "g1k_POPMAX_AF" in hit else None,
-                    'exac_v3_AF': float(hit["exac_AF"] or 0.0) if "exac_AF" in hit else (hit["exac_AC_Adj"]/float(hit["exac_AN_Adj"]) if "exac_AC_Adj" in hit and "exac_AN_Adj"in hit and int(hit["exac_AN_Adj"] or 0) > 0 else None),
-                    'exac_v3_popmax_AF': float(hit["exac_AF_POPMAX"] or 0.0) if "exac_AF_POPMAX" in hit else None,
-                    'gnomad_exomes_AF': float(hit["gnomad_exomes_AF"] or 0.0) if "gnomad_exomes_AF" in hit else None,
-                    'gnomad_exomes_popmax_AF': float(hit["gnomad_exomes_AF_POPMAX_OR_GLOBAL"] or 0.0) if "gnomad_exomes_AF_POPMAX_OR_GLOBAL" in hit else (float(hit["gnomad_exomes_AF_POPMAX"] or 0.0) if "gnomad_exomes_AF_POPMAX" in hit else None),
-                    'gnomad_genomes_AF': float(hit["gnomad_genomes_AF"] or 0.0) if "gnomad_genomes_AF" in hit else None,
-                    'gnomad_genomes_popmax_AF': float(hit["gnomad_genomes_AF_POPMAX_OR_GLOBAL"] or 0.0) if "gnomad_genomes_AF_POPMAX_OR_GLOBAL" in hit else (float(hit["gnomad_genomes_AF_POPMAX"] or 0.0) if "gnomad_genomes_AF_POPMAX" in hit else None),
-                    'topmed_AF': float(hit["topmed_AF"] or 0.0) if "topmed_AF" in hit else None,
-                },
+                'db_freqs': {k: float(hit[k] or 0.0) if k in hit else (0.0 if k else None) for k, v in freq_fields.items()},
```

Comment:
seems to me (0.0 if k else None) should be (0.0 if v is not None else None)

Reply:
or maybe even 
 {k: float(hit[k] or 0.0) if k in hit else 0.0 for k, v in freq_fields.items() if v is not None}

Reply:
no thats the whole point. If there is a key in the mapping but the key is not in the hit we still want to return 0

Reply:
your proposal gets rid of keys entirely when there is nothing present, and the ui expects the key to be there with a null value

Reply:
wait but here you're iterating over `k, v in freq_fields.items()` and saying `0.0 if k else None`, but freq_fields keys are just strings, right?

Reply:
yeah but they are all defined in the form `"AF" if "AF" in index_fields else None` so if `k` is None it means there is no key in the mapping for that field at all, and otherwise there is a key in the mapping, which may or may not be present in the hit

Reply:
even though freq_fields values are defined as `"AF" if "AF" in index_fields else None`, I don't see how k can be None. Like if you have
```
freq_fields = {'1kg_wgs_AF': None,
 '1kg_wgs_popmax_AF': None,
 'AF': None,
 'exac_v3_AF': None,
 'exac_v3_popmax_AF': None}
```
and then do 
```
print([k for k, v in freq_fields.items()])
```

Reply:
oh duh, thanks for catching this